iris_grib/_load_convert.py

```diff
@@ -776,10 +776,35 @@ def grid_definition_template_10(section, metadata):
     # intersects the Earth
     standard_parallel = section["LaD"] * _GRID_ACCURACY_IN_DEGREES
 
+    if "orientationOfTheGrid" in section and not np.isclose(
+        section["orientationOfTheGrid"], 0
+    ):
+        # Could support in future by using the ObliqueMercator class.
+        message = (
+            f"{section['orientationOfTheGrid']=} . iris-grib only supports "
+            "0.0 orientation for grid definition template 10."
+        )
+        raise TranslationError(message)
+
     cs = icoord_systems.Mercator(standard_parallel=standard_parallel, ellipsoid=geog_cs)
 
     # Create the X and Y coordinates.
     x_coord, y_coord, scan = _calculate_proj_coords_from_grid_lengths(section, cs)
+    final_x_point = x_coord.points[-1]
+    final_y_point = y_coord.points[-1]
+    if not (
+        np.isclose(section["longitudeOfLastGridPoint"], final_x_point)
+        and np.isclose(section["latitudeOfLastGridPoint"], final_y_point)
+    ):
```

Comment:
Experiments so far have produced many cases where this block is true, which makes me worried I need an extra factor in here like scaling or units or something. That's why I changed the PR back to draft.

Reply:
Still unsure about this one, but if @pp-mo has no objections I guess I'll take it out of draft.

Reply:
I guess your experiments might indicate that more tolerance is required.
Since it's only a double-check, we can afford to be tolerant.  
Maybe something like atol=0.5 rtol=0.001 ?
So 10. +/- 0.5, 10,000. +/- 10.